vertexai/app/src/main/kotlin/com/google/firebase/quickstart/vertexai/feature/audio/AudioViewModel.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.quickstart.vertexai.feature.audio
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.google.firebase.vertexai.GenerativeModel
+import com.google.firebase.vertexai.type.content
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class AudioViewModel(
+    private val generativeModel: GenerativeModel
+) : ViewModel() {
+
+    private val _uiState: MutableStateFlow<AudioUiState> = MutableStateFlow(AudioUiState.Initial)
+    val uiState: StateFlow<AudioUiState> = _uiState.asStateFlow()
+
+    fun reason(
+        userInput: String,
+        mimeType: String,
+        audioBytes: ByteArray,
+    ) {
+        _uiState.value = AudioUiState.Loading
+        val prompt = if (userInput.isBlank()) "Answer the question in the audio."
+        else "Listen to the audio, and then answer the following question: $userInput"
```

Comment:
We might want to have braces here, for better readability.
From the [Kotlin style guide](https://developer.android.com/kotlin/style-guide#braces):

> Braces are not required for `when` branches and `if` expressions which have no more than one `else` branch and which fit on a single line.
Braces are otherwise required for any `if`, `for`, `when` branch, `do`, and `while` statements and expressions, even when the body is empty or contains only a single statement.

```suggestion
        val prompt = if (userInput.isBlank()) {
          "Answer the question in the audio."
        } else {
          "Listen to the audio, and then answer the following question: $userInput"
        }
```